app/models/user.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+# A class representing a user. This holds information regarding known users and their roles
+# to determine what they should be allowed access to.
+class User < ApplicationRecord
+  enum role: %i[admin instructor]
+  validates_presence_of :utorid
+  validates_uniqueness_of :utorid
```

Comment:
Please have a database constraint for this.

For future reference, you can run `consistency_fail` as a gem in order to find out what unique indices you need to back up your db with based on our models..

Reply:
o good catch